### PR TITLE
fix: [CDS-85388]: Use useGetResolvedTemplate instead of useGetTemplate API to get resolved template

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
### Summary
**Issue:** The Dependent field of the child template did not get resolved properly in the nested template in the pipeline and template studio. 
**Reason:** Nested template did not get resolved in pipeline and template studio so we do not have all the values of the child template so can not populate dependent fields properly.
**Fix:** Use the resolved template in the pipeline and template studio

#### Screenshots

Before: 

The wrong connector type gets populated. (Shell Type: ShellScript so connector type should be WinRM Credential)
Screenshot 2024-01-09 at 5 31 38 PM

Provisioner dropdown not getting populated as the nested template is unresolved.
https://github.com/harness/harness-core-ui/assets/110591046/0e720eff-5fe5-46d3-aea5-fd89ca0e3f63

After: 
Correct connector type: (Shell Type: ShellScript so WinRM Credential)
Screenshot 2024-01-09 at 5 31 45 PM

Provisioner dropdown values getting populated as now the nested template is resolved.

https://github.com/harness/harness-core-ui/assets/110591046/66561b74-411d-4afa-a5aa-790d51dfc83c

Change template in pipeline flow:

https://github.com/harness/harness-core-ui/assets/110591046/5208a46e-ec62-48a1-9f84-ae67cb56c2a2

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by pankajsingh-harness** [comment]
trigger messagecheck

**Review by sonaligoyal-harness — APPROVED** [review]

File: src/modules/72-templates-library/components/PipelineSteps/TemplateStep/TemplateStepWidget/TemplateStepWidget.tsx [hunk 1]
```diff
@@ -112,15 +117,18 @@ function TemplateStepWidget(
 
   React.useEffect(() => {
     setAllValues(
```
Comment (samarth-sharma31) [inline src/modules/72-templates-library/components/PipelineSteps/TemplateStep/TemplateStepWidget/TemplateStepWidget.tsx:119]: Can you please test and attach recording of change template flow - 
Use a stage template in pipeine studio and then change the template from the options inside pipeline studio and link another stage template

Comment (pankajsingh-harness) [reply src/modules/72-templates-library/components/PipelineSteps/TemplateStep/TemplateStepWidget/TemplateStepWidget.tsx:119]: @samarth-sharma31 added the video as request.

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Review by jenil-harness — APPROVED** [review]
LGTM

**Pull request merged**